ibis/backends/clickhouse/compiler.py

```diff
@@ -705,3 +708,31 @@ def visit_TableUnnest(
             )
 
         return select
+
+    def _cleanup_names(
+        self, exprs: Mapping[str, sge.Expression]
+    ) -> Iterator[sge.Expression]:
+        """Compose `_gen_valid_name` and `_dedup_name` to clean up names in projections.
+
+        ClickHouse has a bug where this fails to find the final `"o"."a"` column:
```

Comment:
Is there a link for this bug? If so, might want to add it to the docstring here as a reference.

Reply:
I'll rebase, which should pick up the place where I worked around and linked this bug (#9525)

Reply:
Yep, this should be in #9525.